### ISSUE TITLE
id relationship does not update after the record is provided

### ISSUE THREAD

**Issue by luxferresum** [emberjs/data#5134]
If we load an id-relationship to an not yet loaded record and then later add load that record the belongstoreference is not updated.

```
  async model() {

    // first load foos with id-relations to not yet loaded resources
    // the model has resource: hasMany('resource')
    // and the relationship is resource : { data: { id: 'r1', type: 'resource' } }
    // however resource r1 is NOT sideloaded
    const foos = await this.store.findAll('foo');
    
    // now we load all resources, including all that were referenced by foos
    // this means after this the resource 'r1' is loaded!
    await this.store.findAll('resource');

    // and now the relationship is null on ember-data 2.14.10.
    // this works on ember-data 2.13
    return foos.get('firstObject').belongsTo('resource').value();
  }
```

A git repo to reproduce can be found [here](https://github.com/luxferresum/ember-data-problem-empty-rels).

**Comment by stefanpenner** [comment]
failing test: https://github.com/emberjs/data/pull/5135

**Comment by luxferresum** [comment]
This seems to be fixed on master and 2.16. I haven't checked other versions.
